packages/camera/camera_android_camerax/lib/src/camera_state_error.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// The error that a camera has encountered.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/CameraState.StateError.
+class CameraStateError extends JavaObject {
+  /// Constructs a [CameraStateError] that is not automatically attached to a native object.
+  CameraStateError.detached(
+      {super.binaryMessenger,
+      super.instanceManager,
+      required this.code,
+      required this.description})
+      : super.detached();
+
+  /// The code of this error.
+  final int code;
+
+  /// The description of this error corresponding to its [code].
+  ///
+  /// This is not directly provided by the CameraX library, but is determined on
+  /// the Java side based on the error type.
+  final String description;
+}
+
+/// Flutter API implementation for [CameraStateError].
+///
+/// This class may handle instantiating and adding Dart instances that are
+/// attached to a native instance or receiving callback methods from an
+/// overridden native class.
+@protected
+class CameraStateErrorFlutterApiImpl implements CameraStateErrorFlutterApi {
+  /// Constructs a [CameraStateErrorFlutterApiImpl].
+  CameraStateErrorFlutterApiImpl({
+    this.binaryMessenger,
+    InstanceManager? instanceManager,
+  }) : instanceManager = instanceManager ?? JavaObject.globalInstanceManager;
+
+  /// Receives binary data across the Flutter platform barrier.
+  ///
+  /// If it is null, the default BinaryMessenger will be used which routes to
+  /// the host platform.
+  final BinaryMessenger? binaryMessenger;
```

Comment:
Same question, can these be private?

Reply:
Adding this to list of things to make consistent across the plugin for now.

Reply:
Ok consider this non blocking.